src/app/CMakeLists.txt

```diff
@@ -482,11 +482,8 @@ if(HAVE_OPENCL)
     include_directories(SYSTEM ${OpenCL_INCLUDE_DIRS})
 endif()
 
-if (WITH_PDAL)
-  include_directories(SYSTEM ${PDAL_INCLUDE_DIR})
-  if(MSVC)
-    add_definitions(-DWIN32_LEAN_AND_MEAN)
-  endif()
+if (WITH_PDAL AND MSVC)
+  add_definitions(-DWIN32_LEAN_AND_MEAN)
```

Comment:
I guess this should go to FindPDAL as well with 
[`target_compile_definitions`](https://cmake.org/cmake/help/v3.2/command/target_compile_definitions.html)

Reply:
That would probably not work as the target *is* imported:
> The named <target> must have been created by a command such as [add_executable()](https://cmake.org/cmake/help/v3.2/command/add_executable.html#command:add_executable) or [add_library()](https://cmake.org/cmake/help/v3.2/command/add_library.html#command:add_library) and must not be an [Imported Target](https://cmake.org/cmake/help/v3.2/manual/cmake-buildsystem.7.html#imported-targets).

Reply:
It might work for the fallback code. Wouldn't that part be included with the config-files for Windows?

Reply:
Yes, that should be part of the fallback code or defined by the config files in the Windows case

Reply:
> ...  or defined by the config files in the Windows case

Well, is it?

And is `WIN32_LEAN_AND_MEAN` needed at all nowadays (just looked at this 14-year-old post https://devblogs.microsoft.com/oldnewthing/20091130-00/?p=15863)?
Maybe @jef-n can help us here?

Reply:
it is: https://cdash.orfeo-toolbox.org/build/18183

Reply:
> That would probably not work as the target _is_ imported:
> 
> > The named `<target>` must have been created by a command such as [add_executable()](https://cmake.org/cmake/help/v3.2/command/add_executable.html#command:add_executable) or [add_library()](https://cmake.org/cmake/help/v3.2/command/add_library.html#command:add_library) and must not be an [Imported Target](https://cmake.org/cmake/help/v3.2/manual/cmake-buildsystem.7.html#imported-targets).

That quote was from a very old cmake version docs. If I'm not mistaken, minimal CMake version required for QGIS is 3.12. In that case `target_compile_definitions` seems to work.

Reply:
I added a commit with all windows definitions of WIN32_LEAN_AND_MEAN moved to FindPDAL (using target_compile_definitions).